boxoffice/models/user.py

```diff
@@ -24,14 +27,52 @@ def default_user(context):
     return g.user.id if g.user else None
 
 
+def naive_to_utc(dt, timezone=None):
+    """
+    Returns a UTC datetime for a given naive datetime or date object
+    Localizes it to the given timezone and converts it into a UTC datetime
+    """
+    if timezone:
+        if isinstance(timezone, six.string_types):
+            tz = pytz.timezone(timezone)
+        else:
+            tz = timezone
+    elif isinstance(dt, datetime) and dt.tzinfo:
+        tz = dt.tzinfo
+    else:
+        tz = pytz.UTC
+
+    return tz.localize(dt).astimezone(tz).astimezone(pytz.UTC)
+
+
 class Organization(ProfileBase, db.Model):
     __tablename__ = 'organization'
-    __table_args__ = (db.UniqueConstraint('contact_email'),)
+    __table_args__ = (db.UniqueConstraint('contact_email'),
+        db.CheckConstraint('fy_start_month >= 1 and fy_start_month <= 12', 'org_month_check'),
+        db.CheckConstraint('fy_start_day >= 1 and fy_start_month <= 31', 'org_day_check'))
 
-    # The currently used fields in details are address(html), cin (Corporate Identity Number), pan, service_tax_no, support_email,
+    # The currently used fields in details are address(html), cin (Corporate Identity Number)
+    # pan service_tax_no, support_email,
     # logo (image url), refund_policy (html), ticket_faq (html), website (url)
     details = db.Column(JsonDict, nullable=False, server_default='{}')
     contact_email = db.Column(db.Unicode(254), nullable=False)
+    fy_start_month = db.Column(db.Integer, nullable=True)
+    fy_start_day = db.Column(db.Integer, nullable=True)
```

Comment:
Let's assume the last invoice number in the last FY was 1000 and the last invoice in the current FY is 100. You're running a query for invoices > start_date. If the government announces a date change to Jan 1 and you update it in the database, your query changes to look for the largest invoice number since Jan 1 of this year instead of Apr 1 (as the change becomes effective only _next_ year) and you'll incorrectly generate 1001 instead of 101. You should be recording when the last FY ended, not when the current FY started.

I suggest stepping back and thinking about this a bit. Look at how pytz handles historical time zone changes.